features/support/steps.js

```diff
@@ -25,8 +25,14 @@ Before(function hook() {
 });
 
 After(async function hook() {
-  fs.remove(this.dir);
-  await util.promisify(kill)(this.proc.pid);
+  const pids = [];
+  try {
+    pids.push(...await pidtree(this.proc.pid));
+  } catch (error) {
+    // the process doesn't exist anymore
+  }
+  pids.forEach(pid => process.kill(pid, 'SIGKILL'));
+  return fs.remove(this.dir);
```

Comment:
`fs.remove()` returns a promise, which was previously ignored... the other issues were that:

- tree-kill didn't kill the processes very well
- the directory was removed before the process got killed (caused problems with Ruby hooks)